packages/lib/src/data-grid/DataGrid.tsx

```diff
@@ -0,0 +1,348 @@
+  renderHierarchyTrigger,
+  renderExpandableTrigger,
+} from "./utils";
+import styled, { ThemeProvider } from "styled-components";
+import useTheme from "../useTheme";
+
+const DxcDataGrid = ({
+  columns,
+  rows,
+  selectable,
+  expandable,
+  onSelectRows,
+  selectedRows,
+  uniqueRowId,
+  summaryRow,
+  onGridRowsChange,
+}: DataGridPropsType): JSX.Element => {
+  const [rowsToRender, setRowsToRender] = useState<GridRow[] | HierarchyGridRow[] | ExpandableGridRow[]>(rows);
+  const colorsTheme = useTheme();
+  // Proccess columns prop into usable columns based on other props
+  const columnsToRender = useMemo(() => {
+    let expectedColumns = columns
+      // .filter((column) => visibleColumns.includes(column.name))
+      .map((column) => {
+        // if (!visibleColumns.includes(column.name))
+        //   return {
+        //     key: column.key,
+        //     name: column.name,
+        //     colSpan() {
+        //       return 0;
+        //     },
+        //   };
+
+        return convertToRDGColumns(column, summaryRow);
+      });
+    if (expandable) {
+      expectedColumns = [
+        {
+          key: "expanded",
+          name: "",
+          maxWidth: 36,
+          width: 36,
+          minWidth: 36,
+          colSpan(args) {
+            return args.type === "ROW" && args.row.isExpandedChildContent ? columns.length + 1 : undefined;
+          },
+          renderCell({ row }) {
+            if (row.isExpandedChildContent) {
+              // if it is expanded content
+              return row.expandedChildContent || <></>;
+            }
+            // if row has expandable content
+            return (
+              <ActionContainer id="action">
+                {row.expandedContent && renderExpandableTrigger(row, rowsToRender, uniqueRowId, setRowsToRender)}
+              </ActionContainer>
+            );
+          },
+        },
+        ...expectedColumns,
+      ];
+    }
+    if (!expandable && rows.some((row) => Array.isArray(row.childRows) && row.childRows.length > 0) && uniqueRowId) {
+      // only the first column will be clickable and will expand the rows
+      const firstColumnKey = expectedColumns[0].key;
+      expectedColumns[0] = {
+        ...expectedColumns[0],
+        renderCell({ row }) {
+          if (row.childRows?.length) {
+            return (
+              <HierarchyContainer level={row.rowLevel || 0}>
+                {renderHierarchyTrigger(rowsToRender, row, uniqueRowId, firstColumnKey, setRowsToRender)}
+              </HierarchyContainer>
+            );
+          }
+          return (
+            <HierarchyContainer level={row.rowLevel || 0} className="ellipsis-cell">
+              {row[firstColumnKey]}
+            </HierarchyContainer>
+          );
+        },
+      };
+    }
+    if (selectable) {
+      expectedColumns = [
+        {
+          key: "selected",
+          name: "",
+          maxWidth: 36,
+          width: 36,
+          minWidth: 36,
```

Comment:
Shouldn't this `36` be a token? Something like colorsTheme.dataGrid.columnWidth

Reply:
Unlike our tokens this is a value with no unit and we do not want this value to be edited.